### PR TITLE
Redis client setup

### PR DESCRIPTION
**Pull request by annaxuphoto**
This PR adds a new `RedisClient` class which implements the `Buffer` class to store records in memory, prior to sending them to Firehose. This class borrows heavily from the [`FirehoseClient`](https://github.com/code-dot-org/code-dot-org/blob/staging/lib/cdo/firehose.rb) structure, but uses Redis instead of Firehose to store records. The client stores records in Redis as a list of JSON-ified hashes, and new records are appended to the end of the list.
## Links

- [JIRA](https://codedotorg.atlassian.net/browse/FND-1538?atlOrigin=eyJpIjoiMWU5NjhiMmY0NzQ2NDRlMjk1ZjBmMjVkNDY0YzVkMzEiLCJwIjoiaiJ9)
## Testing story
Wrote unit tests to ensure i18n records were correctly written to Redis

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: lib/cdo/redis.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  }.freeze
+
+  cattr_accessor :client
+  cattr_accessor :log
+
+  # 'The max length of a list is 2^32 - 1 elements (4294967295, more than 4 billion of elements per list).'
+  # Ref: https://redis.io/topics/data-types#:~:text=The%20max%20length%20of%20a%20list%20is%20232%20-%201%20elements%20(4294967295,%20more%20than%204%20billion%20of%20elements%20per%20list).
+  MAX_LIST_LENGTH = 2**32 - 1
+
+  class Buffer < Cdo::Buffer
```
Comment (hacodeorg) [inline lib/cdo/redis.rb]: If the `Buffer` class here and in `FirehoseClient` are mostly the same, can we extract it as a shared class between 2 clients? https://github.com/code-dot-org/code-dot-org/blob/cc9c363c627eb534acd417befa266416fb7a26ea/lib/cdo/firehose.rb#L64

Comment (hacodeorg) [reply lib/cdo/redis.rb]: Or you can give it a different name, `RedisBuffer` for example.

Comment (annaxuphoto) [reply lib/cdo/redis.rb]: Cleaning up the FirehoseClient buffer used to be tracked in the Jira item to swap the two usages, but it's been removed: https://codedotorg.atlassian.net/browse/FND-1539?atlOrigin=eyJpIjoiY2YyMmE3MDk2YWE5NDlmMjk5OTQ4MjYyMzVkMWQ1MGQiLCJwIjoiaiJ9

In the meantime I can rename the buffer class

File: lib/test/cdo/test_redis.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    assert_equal record[:string_key], popped_record["string_key"]
+    assert_equal '"server-side"', popped_record['device']
+  end
+
+  def test_redis_with_multiple_streams
+    record = {
+      url: 'http://studio.code.org/courses',
+      string_key: 'home.heading_elementary_markdown'
+    }
+    RedisClient::STREAMS.each do |stream, _|
```
Comment (hacodeorg) [inline lib/test/cdo/test_redis.rb:48]: Currently this has only 1 stream, right? 
```
  STREAMS = {
    i18n: "i18n-string-tracking-events"
  }.freeze
```

Comment (annaxuphoto) [reply lib/test/cdo/test_redis.rb:48]: yep - I figured leaving the capability for adding more streams was good for future use cases

**Review by hacodeorg — APPROVED** [review]
Thanks for answering my questions  :). The change looks good to me.

**Pull request merged**